packages/nextjs/services/database/config/postgresClient.ts

```diff
@@ -4,26 +4,66 @@ import { drizzle as drizzleNeon } from "drizzle-orm/neon-serverless";
 import { drizzle } from "drizzle-orm/node-postgres";
 import { Pool } from "pg";
 
-let db: ReturnType<typeof drizzle<typeof schema>> | ReturnType<typeof drizzleNeon<typeof schema>>;
-
-const NEON_DB_STRING = "neondb";
-if (process.env.POSTGRES_URL?.includes(NEON_DB_STRING)) {
-  const neonPool = new NeonPool({ connectionString: process.env.POSTGRES_URL as string });
-  db = drizzleNeon(neonPool, { schema, casing: "snake_case" });
-} else {
-  const pool = new Pool({
-    connectionString: process.env.POSTGRES_URL,
-  });
-
-  db = drizzle(pool, {
-    schema,
-    casing: "snake_case",
-  });
-
-  pool.on("error", err => {
-    console.error("Unexpected error on idle client", err);
-    process.exit(-1);
-  });
+type DbInstance = ReturnType<typeof drizzle<typeof schema>> | ReturnType<typeof drizzleNeon<typeof schema>>;
+
+let dbInstance: DbInstance | null = null;
+let poolInstance: Pool | null = null;
+
+function getDb(): DbInstance {
+  if (dbInstance) {
+    return dbInstance;
+  }
+
+  const NEON_DB_STRING = "neondb";
+  if (process.env.POSTGRES_URL?.includes(NEON_DB_STRING)) {
+    const neonPool = new NeonPool({ connectionString: process.env.POSTGRES_URL as string });
```

Comment:
here instead of assining to new variable `neonPool` we assign it to `poolInstance` so that we call end on neon pool instance?

Reply:
I think it makes sense to do that for consistency. 

f2eefde671c0

But, when would you use close in a neon pool? I guess since this is the serverless one, it will never happen, right?

Reply:
Ohh Ohh true!!